tabu/tabu_search.pyx

```diff
@@ -0,0 +1,54 @@
+# distutils: language = c++
+# distutils: include_dirs = tabu/src/
+# distutils: sources = tabu/src/tabu_search.cc
+
+# Copyright 2020 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from libcpp.vector cimport vector
+import numpy as np
+
+cimport tabu
+
+cdef class TabuSearch:
+    """Wraps the class `TabuSearch` from `tabu_search.cc`."""
+    cdef tabu.TabuSearch *c_tabu
+
+    def __cinit__(self, double[:,:] Q, initSol, int tenure, long int timeout, varorder=None) :
```

Comment:
think you want `int[:] initSol`

Reply:
Hmm, I did this because initSol is not always a numpy array..but maybe I should be rewriting how initSol is generated in sampler.py instead

Reply:
another approach would be to do `cdef int[:] initial = np.asarray(initSol, dtype=np.intc)` in the body of the function. Could do the same thing with Q actually which would make it more general.

Reply:
Used dimod.Initialized to generate initial states instead (as suggested by @arcondello), so no need for conversions.